app/src/main/java/org/oppia/android/app/profile/ResetPinDialogFragmentPresenter.kt

```diff
@@ -45,20 +46,20 @@ class ResetPinDialogFragmentPresenter @Inject constructor(
       viewModel = resetViewModel
     }
 
-    binding.inputPin.setInput(resetViewModel.inputPin.get().toString())
-    binding.inputPin.addTextChangedListener(object : TextWatcher {
-      override fun onTextChanged(confirmPin: CharSequence?, start: Int, before: Int, count: Int) {
-        confirmPin?.let {
-          resetViewModel.inputPin.set(confirmPin.toString())
+    // [onTextChanged] is a extension function defined at [TextInputEditTextHelper]
+    binding.resetPinInputPinEditText.onTextChanged { confirmPin ->
+      confirmPin?.let {
+        if (resetViewModel.errorMessage.get() != "" && resetViewModel.inputPin.get() == it) {
+          resetViewModel.inputPin.set(it)
+        } else {
+          resetViewModel.inputPin.set(it)
           resetViewModel.errorMessage.set("")
         }
       }
+    }
```

Comment:
@rt4914 @anandwana001 I've tried this new implementation for the orientation change issue 
as the `onRestoreInstance` solution is not a good approach it requires creating a new variable and it's a bit complicated i've come up with a condition 

here in this condition the error message will not be set to `""` if the error message is not empty and the inputPin is same as the `confirmPin` as this is the only case in which the the `onTextChanged` is triggered after the orientation change

Reply:
for reference
![image](https://user-images.githubusercontent.com/54740946/104004236-82394600-51c9-11eb-9219-e340955833fc.png)

Reply:
Got it. Thanks

Reply:
at place of `resetViewModel.errorMessage.get() != "" ` can we use something like isEmpty or isBlank or any other kotlin function as using `!= ""` doesn't looks good.

Reply:
Done